File: iOS/callvenderapp/callvenderapp.m
//
//  MyClass.m
//  hello
//
//  Created by kisshtink on 12-4-25.
//  Copyright 2012年 __MyCompanyName__. All rights reserved.
//

#import "callvenderapp.h"

@implementation callvenderapp
@synthesize callbackID;
- (id)init
{
    self = [super init];
    if (self) {
        // Initialization code here.
    }
    
    return self;
}

-(void) run:(NSMutableString *)arguments withDict:(NSMutableDictionary *)options
{
    
    //The first argument in the arguments parameter is the callbackID.
    //We use this to send data back to the successCallback or failureCallback
    //through PluginResult.   
    self.callbackID = [arguments pop];
    
    //Get the string that javascript sent us 
    NSString *stringObtainedFromJavascript = [arguments objectAtIndex:0];                 
    NSString *stringObtainedFromJavascript1 = [arguments objectAtIndex:1];                 
    
    //Create the Message that we wish to send to the Javascript
    NSMutableString *stringToReturn = [NSMutableString stringWithString: @"StringReceived:"];
    //Append the received string to the string we plan to send out        
    [stringToReturn appendString: stringObtainedFromJavascript];
    //Create Plugin Result 
    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:                        [stringToReturn stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding]];
    
    NSURL *appUrl = [NSURL URLWithString:stringObtainedFromJavascript ] ; 
    
    NSURL *appUrlStore = [NSURL URLWithString:  stringObtainedFromJavascript1] ; 
    
    
    if([[UIApplication sharedApplication] canOpenURL:appUrl])
    {
         [[UIApplication sharedApplication] openURL:appUrl];
        [self writeJavascript: [pluginResult toSuccessCallbackString:self.callbackID]];
    }
    else
    {
        
        [[UIApplication sharedApplication] openURL:appUrlStore];
        
         [self writeJavascript: [pluginResult toErrorCallbackString:self.callbackID]];
  
    }
    
   
    
}
@end

File: iOS/callvenderapp/callvenderapp.js
/* 
  callvenderapp.js
 

  Created by kissthink on 12-4-26.
  Copyright 2012年 __MyCompanyName__. All rights reserved.
*/


/**
* Phonegap KissthinkPlugin Instance plugin
* Copyright (c)kissthink 2012
*
*/
var KissthinkPlugin = {

run: function(types, success, fail) {
return PhoneGap.exec(success, fail, "callvenderapp", "run", types);
}
};


/**
*   exampel 
*   

KissthinkPlugin.run(
["flypiggroupnote://","http://itunes.apple.com/us/app/.....url..."] ,

function(result) {

alert("Success : \r\n"+result);      

},

function(error) {

alert("Error : \r\n"+error);      
}
); 


*/